src/vector.ts

```diff
@@ -0,0 +1,168 @@
+import { StyleFunction } from 'ol/style/Style';
+import { DOMWidgetModel, ISerializers } from '@jupyter-widgets/base';
+import { MODULE_NAME, MODULE_VERSION } from './version';
+import { Circle as CircleStyle, Fill, Stroke, Style } from 'ol/style.js';
+import { Vector as VectorSource } from 'ol/source.js';
+import { Vector as VectorLayer  } from 'ol/layer.js';
+import { LayerModel, LayerView  } from './layer';
+import 'ol/ol.css';
+import '../css/widget.css';
+
+// -------------------------------------
+import Point    from 'ol/geom/Point.js';
+import Icon     from 'ol/style/Icon.js';
+import Feature  from 'ol/Feature.js';
+import Geometry from 'ol/geom/Geometry';
+// -------------------------------------
+
+export class OpenLayersVectorModel extends LayerModel {
+  defaults() {
+    return {
+      ...super.defaults(),
+      _model_name:           OpenLayersVectorModel.model_name,
+      _model_module:         OpenLayersVectorModel.model_module,
+      _model_module_version: OpenLayersVectorModel.model_module_version,
+      _view_name:            OpenLayersVectorModel.view_name,
+      _view_module:          OpenLayersVectorModel.view_module,
+      _view_module_version:  OpenLayersVectorModel.view_module_version,
+      layers: [],
+    };
+  }
+
+  static serializers: ISerializers = {
+    ...DOMWidgetModel.serializers,
+    // Add any extra serializers here
+  };
+
+  static model_name           = 'OpenLayersVectorModel';
+  static model_module         = MODULE_NAME;
+  static model_module_version = MODULE_VERSION;
+  static view_name            = 'OpenLayersVectorView';
+  static view_module          = MODULE_NAME;
+  static view_module_version  = MODULE_VERSION;
+}
+
+export class OpenLayersVectorView extends LayerView {
+  obj: VectorLayer<any>;
+  render() {
+    this.initVectorLayer();
+    this.create_obj();
+    this.modelEvents();
+  }
+  create_obj() {
+    this.obj = this.vectorLayer;
+  }
+  initVectorLayer() {
+
+    let model_data = this.model.get('data');
+
+    let my_iconFeature = new Feature({
+          geometry: new Point(model_data['geom']),
+          name: 'SPI',
+          population: 4000,
+        });
```

Comment:
Do I understand correctly it supports creating only a single feature?

I'd assume `this.model.get('data')` would be a feature collection.

Reply:
I want to replicate the Point data type as shown in [Icon Symbolizer](https://openlayers.org/en/latest/examples/icon.html) .  So I called the data parameter 'geom' which is a Point(), not a GeoJSON FeatureCollection.

That is, I want to get away from a GeoJSON FeatureCollection, otherwise a single point could have been done with a GeoJSON element (FeatureCollection) with one Feature.  I think it would be difficult to update or delete one icon without changing the entire FeatureCollection.

If there is a second version, there could be multiple Points supplied in a list.  But for now, I wanted to get just a single Point element published.